src/main/java/org/openrewrite/java/testing/assertj/JUnitAssertInstanceOfToAssertThat.java

```diff
@@ -0,0 +1,94 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.List;
+
+public class JUnitAssertInstanceOfToAssertThat extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "JUnit `assertInstanceOf` to AssertJ";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Convert JUnit-style `assertInstanceOf()` to AssertJ's `assertThat().isInstanceOf()`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.junit.jupiter.api.Assertions", false),
+                new AssertInstanceOfToToAssertThatVisitor());
+    }
+
+    public static class AssertInstanceOfToToAssertThatVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private JavaParser.@Nullable Builder<?, ?> assertionsParser;
+
+        private JavaParser.Builder<?, ?> assertionsParser(ExecutionContext ctx) {
+            if (assertionsParser == null) {
+                assertionsParser = JavaParser.fromJavaVersion()
+                        .classpathFromResources(ctx, "assertj-core-3.24");
+            }
+            return assertionsParser;
+        }
+
+        private static final MethodMatcher JUNIT_ASSERT_NULL_MATCHER = new MethodMatcher("org.junit.jupiter.api.Assertions" + " assertInstanceOf(..)");
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            if (!JUNIT_ASSERT_NULL_MATCHER.matches(method)) {
+                return method;
+            }
+
+            List<Expression> args = method.getArguments();
+
+            if (args.size() == 2) {
+                Expression expectedType = args.get(0);
+                Expression actualValue = args.get(1);
+
+                method = JavaTemplate.builder("assertThat(#{any()}).isInstanceOf(#{any()});")
+                        .staticImports("org.assertj.core.api.Assertions.assertThat")
+                        .javaParser(assertionsParser(ctx))
```

Comment:
This should just be this simple; drop the `assertionsParser` field and method; those stemmed from an earlier time when we did not have caching on the JavaParsers created.
```suggestion
                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "assertj-core-3.24"))
```